app/grandchallenge/discussion_forums/templates/discussion_forums/forumtopic_detail.html

```diff
@@ -26,41 +23,27 @@
     </div>
     {% get_obj_perms request.user for object as "user_perms" %}
     <div>
+        {% if 'create_topic_post' in user_perms %}
+            <a hx-get="{% url 'discussion-forums:post-create' slug=object.slug %}" hx-swap="beforeend show:bottom" hx-target="#post-list" class="btn btn-primary btn-sm">
+                <i class="fas fa-comment mr-1"></i> Reply
+            </a>
+        {% endif %}
         {% include 'actstream/partials/follow_unfollow_links.html' with object=object %}
         {% if 'delete_forumtopic' in  user_perms %}
             <a href="{% url 'discussion-forums:topic-delete' slug=object.slug %}" class="btn btn-danger btn-sm">
                 <i class="fas fa-trash mr-1"></i>Delete topic
             </a>
         {% endif %}
     </div>
-    {# TODO: create post button #}
     {# TODO: lock topic button #}
 
     <div class="row">
-        <div class="col-12">
+        <div class="col-12" id="post-list">
             {% for post in object.posts.all %}
-                <div id="{{ post.id }}" class="my-3 card">
-                    <div class="card-body">
-                        <div class="row">
-                            <div class="col-md-10">
-                                <div class="float-right">
-                                    {# TODO: edit button for post #}
-                                    {# TODO: delete button for post #}
-                                </div>
-                                <p>
-                                    <small class="text-muted">
-                                        <i class="fas fa-clock mr-1"></i>By: {{ post.creator|user_profile_link }}
-                                        {{ post.created|naturaltime }}
-                                    </small>
-                                </p>
-                                <div class="post-content">{{ post.content|md2html }}</div>
-                            </div>
-                            <div class="col-md-2 d-none d-md-block">
-                                {% include 'profiles/partials/mugshot.html' with profile=post.creator.user_profile with_userlink=True %}
-                            </div>
-                        </div>
-                    </div>
-                </div>
+                <div class="my-3 card col-12"
+                     id="post-{{ post.pk }}"
+                     hx-get="{% url 'discussion-forums:post-detail' slug=object.slug pk=post.pk %}"
```

Comment:
This is going to create one request per post, similar to an N+1 problem, it would be better if the user does not have to go back to the server each time. Why is each post its own individual view? Do we really need that or can everything be done as part of the topic detail view? If you then need to go to a particular post you could use anchors. Shouldn't the posts also be paginated?

Reply:
The only reason I added the detail view was to be able to easily return the newly created post and swap that into the DOM after post creation, but I can also just return the entire topic detail view again and make sure the entire body is replaced. Then, we can just include each post via a partial and remove the detail view.

Reply:
Maybe it is easiest to turn the `TopicDetail` into a `TopicPostList` view to enable easy pagination.

Reply:
If you use the PaginatedListView the table of topics will update asynchronously without a page refesh as the user goes through the pages. The announcements can be a separate block at the top which is static, would need a separate queryset to get those objects.

Reply:
Ah yeah ok, that might work. it does require a rework of the templates though since currently it's not a table at all. I'll do that in a seperate PR. Easier to get this merged first.

Reply:
I have changed the `TopicDetail` to a list view and added pagination. I will make the changes to the `TopicList` in a seperate PR.

Reply:
True, maybe a normal list view with pagination and an anchor link to the top of the normal topic from page 2 onwards would fit better, otherwise you have a table header and everything.